live-build/misc/migration-scripts/dx_apply

```diff
@@ -16,9 +16,190 @@
+sed -i '/set mainmenu_keycode\[8\]/d' /boot/menu.rc.local
+sed -i '/set mainmenu_command\[8\]/d' /boot/menu.rc.local
+
+#
+# Check that the version to which we are upgrading is compatible with the
+# currently installed version. (Look at version.info to see if we are
+# above the minimum version required. Since we are doing a migration
+# the minimum version should be the latest 5.3.
+#
+# CURRENT_DDS=$(dirname "$(mount | awk '/^\/opt\/delphix /{ print $3 }')")
+# CURRENT_VERSION=$(basename $current_dds)
+#
+# WIP: check if current version is 5.3, if not bail
+#
+
+#
+# Create dataset layout similar to the linux upgrade.
+#
+TMPDIR=$(mktemp -d -p "/tmp" -t delphix.XXXXXXX)
+FSNAME=$(basename "$TMPDIR")
+zfs create -o canmount=off -o mountpoint=none "$RPOOL/ROOT"
+zfs create -o canmount=noauto -o mountpoint=/ "$RPOOL/ROOT/$FSNAME"
+
+#
+# Temporarily set the mountpoint to point to a temporary
+# root directory, and unpack our cpio archive to it.
+#
+TMP_ROOT="$TMPDIR/root"
+zfs set mountpoint="$TMP_ROOT" "$RPOOL/ROOT/$FSNAME"
+zfs mount "$RPOOL/ROOT/$FSNAME"
+(
+	cd "$TMP_ROOT"
+	cpio -imu
+) <"$ARCHIVE_DIR/os-root.cpio" 2>&1 ||
+	die "failed to unpack os-root.cpio"
+
+#
+# WIP: Add here any operations that generate files from illumos,
+#	that we may want to carry over and use in Linux. Examples
+#	include networking interface configuration.
+#
+
+#
+# Set things up for the FreeBSD bootloader to provide an option for
+# booting into the Linux environment This happens in two steps:
+# [1] We place vmlinuz and initrd from our migration image into
+#     the /boot directory of the current system, so the bootloader
+#     can load them.
+# [2] Add a new option in the bootloader's menu for Linux. This is
+#     done by appending the configuration of the new option in our
+#     current menu file.
+#
+VMLINUZ=$(find "$TMP_ROOT/boot/vmlinuz-"* | sort -r | head -n 1)
+INITRD_IMG=$(find "$TMP_ROOT/boot/initrd.img-"* | sort -r | head -n 1)
+cp "$VMLINUZ" /boot
+cp "$INITRD_IMG" /boot
+
+#
+# This command loads the Linux Kernel Compressed Executable in memory
+# from the OK prompt of the FreeBSD bootloader.
+#
+# - We specify the root filesystem type to be ZFS and its root dataset
+#   the one that we just created for the migration image.
+#
+# - We set the console parameter during load twice. Once for each
+#   console technology:
+#      tty0 - Makes kernel messages appear in the first virtual
+#             terminal.
+#      ttyS0 - Makes kernel messages appear in the frst serial
+#             port. The 115200n8 part means the serial connection
+#             is made at 115200 baud 8n1.
+#
+# - We specify zfsforce=1 because otherwise we would be failing at
+#   pool import because of ZFS's hostname check.
+#
+LOAD_VMLINUZ_OK_ARG=(
+	"/boot/$(basename "$VMLINUZ")"
+	"root=ZFS=$RPOOL/ROOT/$FSNAME"
+	'console=tty0 console=ttyS0,115200n8'
+	'zfsforce=1'
+)
+LOAD_VMLINUZ_OK_CMD="load ${LOAD_VMLINUZ_OK_ARG[*]}"
```

Comment:
Any particular reasons for using an array here instead of a regular string? It also seems like quote types `'` and `"` are mixed in the arguments for `LOAD_VMLINUZ_OK_ARG`.